Adding HeatWave support

nugoncal opened the pull request:
- Added HeatWave support
- Added a default mysql_shape_name (which is compatible with HA in case it is enabled)
- Changed 2021 to 2022 in the copyright-related comments

nugoncal committed:
Adding HeatWave support

dralquinta reviewed (changes requested):
README.md should include the new updated format, including a special section for heatwave support. This should include this set of tags at the beginning and end for tool compatibility: 

########## SAMPLE HEATWAVE ENABLED TFVAR FILE ##########

########## SAMPLE HEATWAVE ENABLED TFVAR FILE ##########

In between here, enter the proper expected tfvar file for later usage. 

Under Variable specific considerations, also include that if using `mysql_heatwave_enabled=true`then `mysql_db_system_is_highly_available` must be `false`

Apart from that the code looks good. Can you please amend this to merge to main?

Thanks, 

DA.

nugoncal committed:
Updated README.md

nugoncal commented:
> README.md should include the new updated format, including a special section for heatwave support. This should include this set of tags at the beginning and end for tool compatibility:
> 
> ########## SAMPLE HEATWAVE ENABLED TFVAR FILE ##########
> 
> ########## SAMPLE HEATWAVE ENABLED TFVAR FILE ##########
> 
> In between here, enter the proper expected tfvar file for later usage.
> 
> Under Variable specific considerations, also include that if using `mysql_heatwave_enabled=true`then `mysql_db_system_is_highly_available` must be `false`
> 
> Apart from that the code looks good. Can you please amend this to merge to main?
> 
> Thanks,
> 
> DA.

Addressed the comments and committed.

dralquinta reviewed (approved):
LGTM
